src/keyboard.cpp

```diff
@@ -0,0 +1,155 @@
+#include "keyboard.h"
+
+void Keyboard::notify_key(std::string key) {
+    for(KeyboardListener *listener : listeners) {
+        listener->notify_keyboard_key_pressed(key);
```

Comment:
Nothing to change, but as a note, this could this be really inefficient to have one notification for each key press? What if there was some way we could call `notify_keyboard_key_pressed` one for each listener per update. But premature optimization is bad, so no need to change it now.